lib/campaign/CampaignHandler.cpp

```diff
@@ -588,32 +589,68 @@ CampaignTravel CampaignHandler::readScenarioTravelFromMemory(CBinaryReader & rea
 
 std::vector< std::vector<ui8> > CampaignHandler::getFile(std::unique_ptr<CInputStream> file, const std::string & filename, bool headerOnly)
 {
-	CCompressedStream stream(std::move(file), true);
+	std::vector<ui8> magic(2);
+	file->read(magic.data(), magic.size());
+	file->seek(0);
 
 	std::vector< std::vector<ui8> > ret;
 
-	try
+	if (magic == std::vector<ui8>{0x50, 0x4B}) // VCMP (ZIP)
```

Comment:
1. Create named variable to explain what these magic numbers are
2. Use statically-allocated array instead of vector

```suggestion
	static const std::array<ui8, 2> zipHeaderMagic{0x50, 0x4B};
	if (magic == zipHeaderMagic ) // ZIP archive - assume VCMP format
```